### PR TITLE
python snippets should work on both old and new python versions

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Python 2.x has be sunset already. We should move things to work under both Python 3.x and Python 2.x for some time and then try to remove support for 2.x forever. In this PR, we ensure that the snippets work under both versions as we still use images that only have python2 in them in our CI.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
tested using a WIP PR https://github.com/kubernetes/kubernetes/pull/87654

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
@cblecker this is ready! it's stood up to a lot of testing in https://github.com/kubernetes/kubernetes/pull/87654

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2713,13 +2713,18 @@ function main() {
 
   KUBE_HOME="/home/kubernetes"
   KUBE_BIN=${KUBE_HOME}/bin
-  PYTHON="python"
   CONTAINERIZED_MOUNTER_HOME="${KUBE_HOME}/containerized_mounter"
   PV_RECYCLER_OVERRIDE_TEMPLATE="${KUBE_HOME}/kube-manifests/kubernetes/pv-recycler-template.yaml"
 
-  if [[ "$(python -V)" =~ "Python 3" ]]; then
-    PYTHON="/usr/bin/python2.7"
```
Comment (aojea) [inline cluster/gce/gci/configure-helper.sh]: we don't fall back to python2.7 with the new approach, is that ok?

Comment (dims) [reply cluster/gce/gci/configure-helper.sh]: Yes, this is ok. We want to identify CI jobs that have old python2.7 and need to switch them over

**Comment by neolit123** [comment]
@kubernetes/sig-testing-pr-reviews

**Comment by cheftako** [comment]
Hi @dims, verifying this is correct is a bit complicated and effects some work in progress. Placing a hold while I verify that it isn't going to cause issues. Thank you for the help :)
/hold

**Comment by dims** [comment]
@cheftako "effects some work in progress" sounds intriguing but not enough for a hold, no? 

This has been tested with ubuntu images with python3 (yes on master!) in https://github.com/kubernetes/kubernetes/pull/87654  and is part the effort to be able to validate things like `runc` ( check list here - https://github.com/kubernetes/test-infra/pull/16111#issue-370608240 ). I'd prefer for this to merge and i'll sign up to help with any efforts to improve python3 support that this affects.

**Comment by dims** [comment]
@cheftako i've honored the contract to check thoroughly for python2 before falling back to python3. please take a look

**Comment by dims** [comment]
@BenTheElder @cheftako if this change is enough, please remove hold and lgtm as well :)

File: cluster/gce/gci/configure.sh [hunk 2]
```diff
@@ -66,8 +66,12 @@ function download-kube-env {
     # Convert the yaml format file into a shell-style file.
     eval $(${PYTHON} -c '''
 import pipes,sys,yaml
-for k,v in yaml.load(sys.stdin).iteritems():
-  print("readonly {var}={value}".format(var = k, value = pipes.quote(str(v))))
+if sys.version_info[0] < 3:
```
Comment (cheftako) [inline cluster/gce/gci/configure.sh]: Can we add a quick comment explaining that we are checking the python version and invoking the appropriate versions specific code?

File: cluster/gce/gci/configure-helper.sh [hunk 3]
```diff
@@ -2713,13 +2713,24 @@ function main() {
...
-  PYTHON="python"
   CONTAINERIZED_MOUNTER_HOME="${KUBE_HOME}/containerized_mounter"
   PV_RECYCLER_OVERRIDE_TEMPLATE="${KUBE_HOME}/kube-manifests/kubernetes/pv-recycler-template.yaml"
 
-  if [[ "$(python -V)" =~ "Python 3" ]]; then
+  if [[ "$(python -V)" =~ "Python 2" ]]; then
+    # found python2, just use that
+    PYTHON="python"
+  elif [[ -d "/usr/bin/python2.7" ]]; then
+    # found python2 in a different directory
```
Comment (cheftako) [inline cluster/gce/gci/configure-helper.sh]: # System python not defaulted to 2 but using 2.7 during migration

File: cluster/gce/gci/configure.sh (hunk 2 above)
Comment (dims) [reply cluster/gce/gci/configure.sh]: Doing it now

File: cluster/gce/gci/configure-helper.sh (hunk 3 above)
Comment (dims) [reply cluster/gce/gci/configure-helper.sh]: Doing that now

**Comment by dims** [comment]
Fixed both things that you mentioned @cheftako thanks!

**Comment by dims** [comment]
/hold cancel

( as i see a /lgtm from @cheftako )

**Comment by dims** [comment]
@cheftako @BenTheElder do you mind reapplying `/lgtm`?

**Review by cheftako — COMMENTED** [review]
/lgtm

**Pull request merged**